arch/csr/hstateen0.yaml

```diff
@@ -0,0 +1,201 @@
+        return 0;
+      }
+  IMSIC:
+    long_name: IMSIC state access control
+    location: 58
+    definedBy: Ssaia
+    base: 64
+    description: |
+      The IMSIC bit in `hstateen0` controls access to the guest IMSIC state,
+      including CSRs `stopei` (really `vstopei`), provided by the Ssaia extension.
+
+      Setting the IMSIC bit in `hstateen0` to zero prevents a virtual machine
+      from accessing the hart’s IMSIC the same as setting `hstatus.`VGEIN = 0.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+    sw_write(csr_value): |
+      if (CSR[mstateen0].IMSIC == 1'b0){
+        return 0;
+      }
+  CONTEXT:
+    long_name: scontext access control
+    location: 57
+    definedBy: Sdtrig
+    base: 64
+    description: |
+      The CONTEXT bit in `hstateen0` controls access to the `scontext` CSR provided
+      by the Sdtrig extension.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+    sw_write(csr_value): |
+      if (CSR[mstateen0].CONTEXT == 1'b0){
+        return 0;
+      }
+  JVT:
+    long_name: jvt access control
+    location: 2
+    definedBy: Zcmt
+    description: |
+      The JVT bit controls access to the `jvt` CSR provided by the Zcmt extension.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+    sw_write(csr_value): |
+      if (CSR[mstateen0].JVT == 1'b0){
+        return 0;
+      }
+  FCSR:
+    long_name: fcsr access control
+    location: 1
+    #definedBy:
+    #anyOf: [Zfinx, Zdinx]
+    description: |
+      The FCSR bit controls access to `fcsr` for the case when floating-point instructions
+      operate on `x` registers instead of `f` registers as specified by the Zfinx and related
+      extensions (Zdinx, etc.). Whenever `misa.F` = 1, FCSR bit of `mstateen0` is read-only
+      zero (and hence read-only zero in `hstateen0` and `sstateen0` too). For convenience,
+      when the `stateen` CSRs are implemented and `misa.F` = 0, then if the FCSR bit of a
+      controlling `stateen0` CSR is zero, all floating-point instructions cause an illegal
+      instruction trap (or virtual instruction trap, if relevant), as though they all access
+      `fcsr`, regardless of whether they really do.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+    sw_write(csr_value): |
+      if (CSR[mstateen0].FCSR == 1'b0){
+        return 0;
+      }
+  C:
+    long_name: custom state access control
+    location: 0
+    description: |
+      The C bit controls access to any and all custom state. The C bit of these registers is
+      not custom state itself; it is a standard field of a standard CSR, either `mstateen0`,
+      `hstateen0`, or `sstateen0`.
+    type: RW
+    reset_value: UNDEFINED_LEGAL
+    sw_write(csr_value): |
+      if (CSR[mstateen0].C == 1'b0){
+        return 0;
+      }
+sw_read(): |
+  # for every bit in an mstateen CSR that is zero, the same bit
+  # appears as read-only zero in the matching hstateen CSR
+
+  Bits<64> mstateen0_mask = CSR[mstateen0].csr_value;
```

Comment:
The CI failures are complaining about this. Looking at other files, when the entire CSR value is accessed, it seems you should use:
```
$bits(CSR[mstateen0])
```
here and elsewhere